File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/ClazzActivityChange.java
package com.ustadmobile.lib.db.entities;

import com.ustadmobile.lib.database.annotation.UmEntity;
import com.ustadmobile.lib.database.annotation.UmPrimaryKey;
import com.ustadmobile.lib.database.annotation.UmSyncLastChangedBy;
import com.ustadmobile.lib.database.annotation.UmSyncLocalChangeSeqNum;
import com.ustadmobile.lib.database.annotation.UmSyncMasterChangeSeqNum;

@UmEntity(tableId = 32)
public class ClazzActivityChange{

    public static final int UOM_FREQUENCY = 1;
    public static final int UOM_DURATION = 2;
    public static final int UOM_BINARY = 3;

    @UmPrimaryKey(autoGenerateSyncable = true)
    private long clazzActivityChangeUid;

    private String clazzActivityChangeTitle;

    private String clazzActivityDesc;

    private int clazzActivityUnitOfMeasure;

    private boolean clazzActivityChangeActive;

    public int getClazzActivityChangeLastChangedBy() {
        return clazzActivityChangeLastChangedBy;
    }

    public void setClazzActivityChangeLastChangedBy(int clazzActivityChangeLastChangedBy) {
        this.clazzActivityChangeLastChangedBy = clazzActivityChangeLastChangedBy;
    }

    @UmSyncLastChangedBy
    private int clazzActivityChangeLastChangedBy;



    public boolean isClazzActivityChangeActive() {
        return clazzActivityChangeActive;
    }

    public void setClazzActivityChangeActive(boolean clazzActivityChangeActive) {
        this.clazzActivityChangeActive = clazzActivityChangeActive;
    }

    @UmSyncMasterChangeSeqNum
    private long clazzActivityChangeMasterChangeSeqNum;

    @UmSyncLocalChangeSeqNum
    private long clazzActivityChangeLocalChangeSeqNum;

    @UmSyncLastChangedBy
    private int clazzActivityLastChangedBy;

    public int getClazzActivityUnitOfMeasure() {
        return clazzActivityUnitOfMeasure;
    }

    public void setClazzActivityUnitOfMeasure(int clazzActivityUnitOfMeasure) {
        this.clazzActivityUnitOfMeasure = clazzActivityUnitOfMeasure;
    }

    public long getClazzActivityChangeUid() {
        return clazzActivityChangeUid;
    }

    public void setClazzActivityChangeUid(long clazzActivityChangeUid) {
        this.clazzActivityChangeUid = clazzActivityChangeUid;
    }

    public String getClazzActivityChangeTitle() {
        return clazzActivityChangeTitle;
    }

    public void setClazzActivityChangeTitle(String clazzActivityTitle) {
        this.clazzActivityChangeTitle = clazzActivityTitle;
    }

    public String getClazzActivityDesc() {
        return clazzActivityDesc;
    }

    public void setClazzActivityDesc(String clazzActivityDesc) {
        this.clazzActivityDesc = clazzActivityDesc;
    }

    public long getClazzActivityChangeMasterChangeSeqNum() {
        return clazzActivityChangeMasterChangeSeqNum;
    }

    public void setClazzActivityChangeMasterChangeSeqNum(long clazzActivityChangeMasterChangeSeqNum) {
        this.clazzActivityChangeMasterChangeSeqNum = clazzActivityChangeMasterChangeSeqNum;
    }

    public long getClazzActivityChangeLocalChangeSeqNum() {
        return clazzActivityChangeLocalChangeSeqNum;
    }

    public void setClazzActivityChangeLocalChangeSeqNum(long clazzActivityChangeLocalChangeSeqNum) {
        this.clazzActivityChangeLocalChangeSeqNum = clazzActivityChangeLocalChangeSeqNum;
    }

    public int getClazzActivityLastChangedBy() {
        return clazzActivityLastChangedBy;
    }

    public void setClazzActivityLastChangedBy(int clazzActivityLastChangedBy) {
        this.clazzActivityLastChangedBy = clazzActivityLastChangedBy;
    }
}

File: lib-database-entities/src/commonMain/kotlin/com/ustadmobile/lib/db/entities/ClazzActivityChange.kt
package com.ustadmobile.lib.db.entities

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.ustadmobile.door.annotation.LastChangedBy
import com.ustadmobile.door.annotation.LocalChangeSeqNum
import com.ustadmobile.door.annotation.MasterChangeSeqNum
import com.ustadmobile.door.annotation.SyncableEntity
import kotlinx.serialization.Serializable


@SyncableEntity(tableId = 32)
@Entity
@Serializable
class ClazzActivityChange {

    @PrimaryKey(autoGenerate = true)
    var clazzActivityChangeUid: Long = 0

    var clazzActivityChangeTitle: String? = null

    var clazzActivityDesc: String? = null

    var clazzActivityUnitOfMeasure: Int = 0

    var isClazzActivityChangeActive: Boolean = false

    @LastChangedBy
    var clazzActivityChangeLastChangedBy: Int = 0

    @MasterChangeSeqNum
    var clazzActivityChangeMasterChangeSeqNum: Long = 0

    @LocalChangeSeqNum
    var clazzActivityChangeLocalChangeSeqNum: Long = 0

    @LastChangedBy
    var clazzActivityLastChangedBy: Int = 0

    companion object {

        val UOM_FREQUENCY = 1
        val UOM_DURATION = 2
        val UOM_BINARY = 3
    }
}
